### PR TITLE
Feature Bits for 2017 HCAL HF TP

### PR DESCRIPTION
**Pull request by matz-e**
Sits on top of #15945.
- Refactors the channel validity check to happen earlier
- Port of the 2016 MinBias feature bit
- Refactor the EM feature bit and hook it up to 2017 TP (if manually configured in the PSet)

### CODE REVIEW

**Comment by kpedro88** [comment]
@cmsbuild please test

**Comment by matz-e** [comment]
Again, @cmsbuild says comparisons are ready, but the linked page shows them as "Running"…

**Comment by kpedro88** [comment]
@cmsbuild please test

**Comment by kpedro88** [comment]
@rekovic, @mulhearn comparisons finally finished, I don't observe any discrepancies

**Comment by mulhearn** [comment]
+1

File: SimCalorimetry/HcalTrigPrimAlgos/src/HcalTriggerPrimitiveAlgo.cc [hunk 1]
```diff
@@ -513,9 +504,31 @@ void HcalTriggerPrimitiveAlgo::analyzeHF2016(
...
+{
+   auto mask = conditions_->getHcalTPChannelParameter(HcalDetId(digi.id()))->getMask();
+   if (mask)
+      return false;
+
+   auto parameters = conditions_->getHcalTPParameters();
+   auto adc_threshold = parameters->getADCThresholdHF();
+   auto tdc_mask = parameters->getTDCMaskHF();
+
+   if (override_parameters_.exists("ADCThresholdHF"))
```
Comment (davidlange6) [inline SimCalorimetry/HcalTrigPrimAlgos/src/HcalTriggerPrimitiveAlgo.cc]: Hi @matz-e - we can't do this sort of check for every digi. can you see who to move it to a once per job check. Thanks

File: SimCalorimetry/HcalTrigPrimAlgos/src/HcalTriggerPrimitiveAlgo.cc [hunk 2]
```diff
@@ -586,19 +593,31 @@ void HcalTriggerPrimitiveAlgo::analyzeHF2017(
...
             }
 
-            // int ADCLong = details.LongDigi[ibin].adc();
-            // int ADCShort = details.ShortDigi[ibin].adc();
-            // if(hcalfem != 0)
-            // {
-            //     finegrain[ibin] = (finegrain[ibin] || hcalfem->fineGrainbit(ADCShort, details.ShortDigi.id(), details.ShortDigi[ibin].capid(), ADCLong, details.LongDigi.id(), details.LongDigi[ibin].capid()));
-            // }
+            for (const auto& detail: details) {
+               if (detail.validity[idx] and HcalDetId(detail.digi.id()).ietaAbs() != 29) {
```
Comment (davidlange6) [inline SimCalorimetry/HcalTrigPrimAlgos/src/HcalTriggerPrimitiveAlgo.cc]: please document the magic number 29

**Comment by kpedro88** [comment]
@cmsbuild please test

**Comment by matz-e** [comment]
Ping @mulhearn, @rekovic for re-approval?

**Pull request merged**